### PR TITLE
remove testify dependency

### PR DESCRIPTION
**Pull request by binjip978**
Signed-off-by: Sergei Semenchuk <pdp.eleven11@gmail.com>

### CODE REVIEW

**Comment by binjip978** [comment]
fixes https://github.com/u-root/u-root/issues/2187
but not sure about ARM test and two test are failing that currently disabled

**Review by rminnich — COMMENTED** [review]
This is wonderful!

While the error messages are not in the recommended format described here:
https://github.com/u-root/u-root/blob/main/CONTRIBUTING.md#unit-testing-guidelines

They are no worse than what we have now, so I'm inclined to approve this.

If you are up for changing the messages to follow the recommended format, that would be very nice, else this can go in as is, once the build failure is fixed.

**Comment by binjip978** [comment]
> This is wonderful!
> 
> While the error messages are not in the recommended format described here: https://github.com/u-root/u-root/blob/main/CONTRIBUTING.md#unit-testing-guidelines
> 
> They are no worse than what we have now, so I'm inclined to approve this.
> 
> If you are up for changing the messages to follow the recommended format, that would be very nice, else this can go in as is, once the build failure is fixed.

Sure will do formatting.

**Comment by binjip978** [comment]
@rminnich updated with recommended format, probably missed something, but anyway should look better

**Review by rminnich — DISMISSED** [review]
WOW!

File: pkg/crypto/ed25519_test.go [hunk 1]
```diff
@@ -33,77 +32,113 @@ var password = []byte{'k', 'e', 'i', 'n', 's'}
...
+		t.Errorf(`ed25519.Verify(publicKey, testData, signatureBad) = %t, want "false"`, verified)
+	}
 }
 
 func TestGenerateKeys(t *testing.T) {
 	tmpdir := t.TempDir()
 
 	err := GeneratED25519Key(password, path.Join(tmpdir, "private_key.pem"), path.Join(tmpdir, "public_key.pem"))
-	require.NoError(t, err)
+	if err != nil {
```
Comment (rjoleary) [inline pkg/crypto/ed25519_test.go:132]: if err := ...; err != nil {

File: pkg/crypto/ed25519_test.go [hunk 2]
```diff
@@ -33,77 +32,113 @@ var password = []byte{'k', 'e', 'i', 'n', 's'}
...
+		t.Errorf(`GeneratED25519Key(password, path.Join(tmpdir, "private_key.pem"), path.Join(tmpdir, "public_key.pem")) = %v, want nil`, err)
+	}
 }
 
 func TestGenerateUnprotectedKeys(t *testing.T) {
 	tmpdir := t.TempDir()
 
 	err := GeneratED25519Key(nil, path.Join(tmpdir, "private_key.pem"), path.Join(tmpdir, "public_key.pem"))
-	require.NoError(t, err)
+	if err != nil {
```
Comment (rjoleary) [inline pkg/crypto/ed25519_test.go:141]: if err := ...; err != nil {

File: pkg/mount/block/blockdev_test.go [hunk 3]
```diff
@@ -19,16 +18,22 @@ import (
 func TestFindMountPointNotExists(t *testing.T) {
 	LinuxMountsPath = "testdata/mounts"
 	_, err := GetMountpointByDevice("/dev/mapper/sys-oldxxxxxx")
-	require.Error(t, err)
+	if err == nil {
```
Comment (rjoleary) [inline pkg/mount/block/blockdev_test.go:21]: if err := ...; err != nil {

File: pkg/mount/block/blockdev_test.go [hunk 4]
```diff
@@ -105,7 +110,7 @@ func TestParsePCIBlockList(t *testing.T) {
 				for _, d := range tt.want {
 					s = fmt.Sprintf("%s{Vendor: %v, Device %v}\n", s, d.Vendor, d.Device)
 				}
-				t.Error(s)
+				t.Errorf(`reflect.DeepEqual(%v, %v) = false, want true`, devices, tt.want)
```
Comment (rjoleary) [inline pkg/mount/block/blockdev_test.go:113]: Use " quotes instead of `

**Review by rjoleary — COMMENTED** [review]
This looks awesome! It has to be rebased onto main.

**Comment by binjip978** [comment]
> This looks awesome! It has to be rebased onto main.

rebased, but not sure if moving everything to `if err := ...; err != nil {` is valuable, because some lines are already too long and it already too much of error checking in this PR.

File: pkg/mount/block/blockdev_test.go (hunk 4 above)
Comment (binjip978) [reply pkg/mount/block/blockdev_test.go]: updated

File: pkg/mount/block/blockdev_test.go (hunk 3 above)
Comment (binjip978) [reply pkg/mount/block/blockdev_test.go]: updated

File: pkg/crypto/ed25519_test.go (hunk 2 above)
Comment (binjip978) [reply pkg/crypto/ed25519_test.go]: updated

File: pkg/crypto/ed25519_test.go (hunk 1 above)
Comment (binjip978) [reply pkg/crypto/ed25519_test.go]: updated

File: tools/vpdbootmanager/get_test.go [hunk 5]
```diff
@@ -8,40 +8,51 @@ import (
...
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 	"github.com/u-root/u-root/pkg/vpd"
 )
 
 var getter *Getter
 
-func testNewgetter(t *testing.T) {
+func TestNewgetter(t *testing.T) {
 	g := NewGetter()
```
Comment (rjoleary) [inline tools/vpdbootmanager/get_test.go:17]:
```
	if g := NewGetter(); g == nil {
		t.Fatalf(`NewGetter() = %v, want not nil`, g)
	} else if g.R == nil || g.Out == nil {
		t.Errorf(`g.R, g.Out = %v, %v, want not nil, not nil`, g.R, g.Out)
	}
```

File: tools/vpdbootmanager/get_test.go [hunk 6]
```diff
@@ -8,40 +8,51 @@ import (
...
 	var buf bytes.Buffer
 	getter := NewGetter()
 	getter.Out = &buf
 	getter.R = vpd.NewReader()
 	getter.R.VpdDir = "./tests"
 	err := getter.Print("firmware_version")
-	require.NoError(t, err)
-	out := buf.String()
-	assert.Equal(t, "firmware_version(RO) => 1.2.3\n\nfirmware_version(RW) => 3.2.1\n\n", out)
+	if err != nil {
```
Comment (rjoleary) [inline tools/vpdbootmanager/get_test.go:35]: if err := ...; err != nil {

File: tools/vpdbootmanager/main_test.go [hunk 7]
```diff
@@ -10,63 +10,91 @@ import (
...
 
-	"github.com/stretchr/testify/require"
 	"github.com/u-root/u-root/pkg/boot/systembooter"
 )
 
-func testInvalidCommand(t *testing.T) {
+func TestInvalidCommand(t *testing.T) {
 	err := cli([]string{"unknown"})
-	require.Equal(t, "Unrecognized action", err.Error())
+	if err.Error() != "Unrecognized action" {
```
Comment (rjoleary) [inline tools/vpdbootmanager/main_test.go:18]: if err := cli([]string{"unknown"}; err.Error() != "Unrecognized action" {

File: tools/vpdbootmanager/main_test.go [hunk 8]
```diff
@@ -10,63 +10,91 @@ import (
...
+		t.Errorf(`err.Error() = %q, want "you need to provide action"`, err.Error())
+	}
 }
 
-func testAddNetbootEntryFull(t *testing.T) {
+func TestAddNetbootEntryFull(t *testing.T) {
 	dir := t.TempDir()
-	os.MkdirAll(path.Join(dir, "rw"), 0o700)
+	err := os.MkdirAll(path.Join(dir, "rw"), 0o700)
+	if err != nil {
```
Comment (rjoleary) [inline tools/vpdbootmanager/main_test.go:40]: if err := ...; err != nil {

File: tools/vpdbootmanager/main_test.go [hunk 9]
```diff
@@ -10,63 +10,91 @@ import (
...
 		"netboot",
 		"dhcpv6",
 		"aa:bb:cc:dd:ee:ff",
 		"-vpd-dir",
 		dir,
-	})
-	require.NoError(t, err)
+	}
+	err = cli(args)
+	if err != nil {
```
Comment (rjoleary) [inline tools/vpdbootmanager/main_test.go:53]: if err := ...; err != nil {

File: tools/vpdbootmanager/main_test.go [hunk 10]
```diff
@@ -10,63 +10,91 @@ import (
...
-	require.NoError(t, err)
+	if err != nil {
+		t.Errorf(`os.ReadFile(path.Join(%q, "rw", "Boot0001")) = %v, want nil`, file, err)
+	}
 	var out systembooter.NetBooter
 	err = json.Unmarshal([]byte(file), &out)
-	require.NoError(t, err)
-	require.Equal(t, "dhcpv6", out.Method)
-	require.Equal(t, "aa:bb:cc:dd:ee:ff", out.MAC)
+	if err != nil {
```
Comment (rjoleary) [inline tools/vpdbootmanager/main_test.go:62]: if err := ...; err != nil {

**Review by rjoleary — COMMENTED** [review]
It looks like this also gets rid of the yaml.v3 dependency which is great! Can you rebase again?

**Comment by binjip978** [comment]
> It looks like this also gets rid of the yaml.v3 dependency which is great! Can you rebase again?

rebased, let probably merge that if it ok, and I will make additional PR with all of this style issues to avoid rebasing it after any other commit

**Review by rjoleary — APPROVED** [review]

**Pull request merged**